src/runtime/constructorbinder.cs

```diff
@@ -51,10 +51,15 @@ internal object InvokeRaw(IntPtr inst, IntPtr args, IntPtr kw)
         /// </remarks>
         internal object InvokeRaw(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info)
         {
+            if (!_containingType.Valid)
+            {
+                return Exceptions.RaiseTypeError(_containingType.DeletedMessage);
+            }
             object result;
+            Type tp = _containingType.Value;
 
-            if (_containingType.IsValueType && !_containingType.IsPrimitive &&
-                !_containingType.IsEnum && _containingType != typeof(decimal) &&
+            if (tp.IsValueType && !tp.IsPrimitive &&
+                !tp.IsEnum && tp != typeof(decimal) &&
```

Comment:
NIT: I wonder if we could reduce diff significantly by using the following pattern:

1. Rename fields, that were changed to `Maybe*`
2. Start methods, that use that member with `if (!field.TryGetValue(out var originalFieldName)) return Exceptions.Raise...;`

E.g.

```csharp
private MaybeType parentType; // <- creative renaming here

object InvokeRaw(...)
{
  if (!parentType.TryGet(out var _containingType)) return Exceptions.Raise(...);
  if (_containingType.IsValueType && !_containingType.IsPrimitive && // <- this and below are now untouched
```

Reply:
I don't think we can. Most are static methods using a class instance as a parameter to operate on like `self.type.<member or method>`. 

We could implement those members or methods on the `Maybe*`, but every time a method/member that wasn't "exposed" is used, we'd have to add it. And it might make people to forget to check validity before use.